### PR TITLE
[ACCESS-2693] Update api spec with permissions information

### PR DESCRIPTION
**Pull request by wangwillson1**
### What does this PR do? What is the motivation?

In the API reference, this PR adds the permissions required to access a given endpoint.

https://docs-staging.datadoghq.com/willson.wang/ACCESS-2693-update-api-spec/api/latest/users/#update-a-user

### Merge instructions

- Please wait for a review from aaa-core-access before merging

### CODE REVIEW

**Review by srosenthal-dd — COMMENTED** [review]
Can you help me understand what this changes means from the user perspective?

Will all API endpoints that start to list permissions in the public docs?

I assume the extra OAuth section is because not all permissions are available in OAuth? I do hope that is a temporary situation in most cases.

We (including PM) should be aware that the actual permission logic is occasionally more complicated. In the example you link, "update a user", the first permission check is an OR, but then it looks specifically at the user/service-account being updated and validates that the caller has the appropriate permission ([code](https://github.com/DataDog/dogweb/blob/4f45b5b47b0f4a8d1f82957c8bf948bee57e3832/dogweb/controllers/api/user.py#L1021))

**Comment by wangwillson1** [comment]
> Will all API endpoints that start to list permissions in the public docs?

all non-rapid endpoints, since API clients is working on adding api spec autogen support for them in the coming weeks

> I assume the extra OAuth section is because not all permissions are available in OAuth? I do hope that is a temporary situation in most cases.

yes precisely - ideally we can (one day) just get rid of the whole notion of separate oauth endpoints, but once we get there the docs change will be relatively easy

**Review by bi1yeu — APPROVED** [review]

File: layouts/partials/api/permissions.html [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      <li><code>{{ . }}</code></li>
+    {{ end }}
+    <br/>
+  {{ end }}
+{{ end }}
+
+
+{{ with $authZ }}
+  {{ $slug := (replaceRE " " "-" ($tag.name | humanize | lower)) }}
+  <p>OAuth apps will require the <code>{{ delimit . ", "}}</code> authorization <a href="{{ "api/latest/scopes/" | absLangURL }}#{{ $slug }}">scope</a> to access this endpoint.</p>
```
Comment (urseberry) [inline layouts/partials/api/permissions.html:40]:
```suggestion
  <p>OAuth apps require the <code>{{ delimit . ", "}}</code> authorization <a href="{{ "api/latest/scopes/" | absLangURL }}#{{ $slug }}">scope</a> to access this endpoint.</p>
```
We use present tense in the public docs.

**Review by bgdeutsch — APPROVED** [review]

**Review by urseberry — APPROVED** [review]

**Comment by wangwillson1** [comment]
> Willson, do you want me to merge this, or should I wait for Stephen to re-review?

good to merge now, thank you!

**Review by urseberry — APPROVED** [review]

**Pull request merged**